### PR TITLE
fix: private team does not hide team members from event type page

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

- Fixes https://app.campsite.com/cal/posts/nx8j3hx4btuw
- Fixes CAL-4392 (Linear issue number - should be visible at the bottom of the GitHub issue description)

Before:-

When a team is private, a team member can see all assigned users for the team event type and can also be redirected to their public page

Screenshot 2024-10-15 002714
Screenshot 2024-10-15 001914

After:-

Screenshot 2024-10-15 001930

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A-I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] N/A-I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

**Review by zomars — CHANGES_REQUESTED** [review]
Can we add a before and after? Let's at least fill the PR requirements (how to test and so on...) 🙏

File: apps/web/modules/event-types/views/event-types-listing-view.tsx [hunk 1]
```diff
@@ -260,6 +260,13 @@ export const InfiniteEventTypeList = ({
     },
   });
 
+  const { data: team } = trpc.viewer.teams.get.useQuery({ teamId: group.teamId ?? -1 });
+
+  const isTeamAdminOrOwner =
+    team && (team.membership.role === MembershipRole.OWNER || team.membership.role === MembershipRole.ADMIN);
+
+  const isTeamPrivate = !!(team?.isPrivate && !isTeamAdminOrOwner);
```
Comment (zomars) [inline apps/web/modules/event-types/views/event-types-listing-view.tsx:268]: I don't think this is enough to protect the privacy. We're doing this stuff on the client. Which any member can inspect the HTML and check the NEXT_DATA in the json. Getting all users and emails. Which I don't think is desired. We should instead incorporate these changes in `trpc.viewer.eventTypes.getEventTypesFromGroup` query,

**Review by zomars — DISMISSED** [review]
Left some more feedback 🙏🏽

File: apps/web/modules/event-types/views/event-types-listing-view.tsx (hunk 1 above)
Comment (anikdhabal) [reply apps/web/modules/event-types/views/event-types-listing-view.tsx]: Thanks for the explanation, completely agree with you. Made changes🙏🏽

File: packages/trpc/server/routers/viewer/eventTypes/getEventTypesFromGroup.handler.ts [hunk 2]
```diff
@@ -118,6 +119,27 @@ export const getEventTypesFromGroup = async ({ ctx, input }: GetByViewerOptions)
...
+          isPrivate: true,
+        },
+      },
+    },
+  });
+
+  if (membership && membership.team.isPrivate)
+    filteredEventTypes.forEach((evType) => {
+      evType.users = [];
+    });
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/viewer/eventTypes/getEventTypesFromGroup.handler.ts:141]: There is also a hosts property that returns users

**Review by Praashh — APPROVED** [review]

**Pull request merged**